tests/github/experimental/workflow/FederatedRuntime/testcase_include_exclude/workspace/testflow_include_exclude.ipynb

```diff
@@ -0,0 +1,364 @@
+    "    WARNING = \"\\033[93m\"\n",
+    "    FAIL = \"\\033[91m\"\n",
+    "    ENDC = \"\\033[0m\"\n",
+    "    BOLD = \"\\033[1m\"\n",
+    "    UNDERLINE = \"\\033[4m\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "36ed5e31",
+   "metadata": {},
+   "source": [
+    "Let us now define the flow of the testcase include_exclude"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "52c4a752",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| export\n",
+    "\n",
+    "class TestFlowIncludeExclude(FLSpec):\n",
+    "    \"\"\"\n",
+    "    Testflow to validate include and exclude functionality in Federated Flow.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    include_exclude_error_list = []\n",
+    "\n",
+    "    def __init__(self, checkpoint: bool = False):\n",
+    "        super().__init__(checkpoint)\n",
+    "\n",
+    "    @aggregator\n",
+    "    def start(self):\n",
+    "        \"\"\"\n",
+    "        Flow start.\n",
+    "        \"\"\"\n",
+    "        print(\n",
+    "            f\"{bcolors.OKBLUE}Testing FederatedFlow - Starting Test for Include and Exclude \"\n",
+    "            + f\"Attributes {bcolors.ENDC}\"\n",
+    "        )\n",
+    "        self.collaborators = self.runtime.collaborators\n",
+    "\n",
+    "        self.exclude_agg_to_agg = 10\n",
+    "        self.include_agg_to_agg = 100\n",
+    "        self.next(self.test_include_exclude_agg_to_agg, exclude=[\"exclude_agg_to_agg\"])\n",
+    "\n",
+    "    @aggregator\n",
+    "    def test_include_exclude_agg_to_agg(self):\n",
+    "        \"\"\"\n",
+    "        Testing whether attributes are excluded from agg to agg\n",
+    "        \"\"\"\n",
+    "        if (\n",
+    "            hasattr(self, \"include_agg_to_agg\") is True\n",
+    "            and hasattr(self, \"exclude_agg_to_agg\") is False\n",
+    "        ):\n",
+    "            print(\n",
+    "                f\"{bcolors.OKGREEN} ... Exclude test passed in test_include_exclude_agg_to_agg \"\n",
+    "                + f\"{bcolors.ENDC}\"\n",
+    "            )\n",
+    "        else:\n",
+    "            TestFlowIncludeExclude.include_exclude_error_list.append(\n",
+    "                \"test_include_exclude_agg_to_agg\"\n",
+    "            )\n",
+    "            print(\n",
+    "                f\"{bcolors.FAIL} ... Exclude test failed in test_incude_exclude_agg_to_agg \"\n",
+    "                + f\"{bcolors.ENDC}\"\n",
+    "            )\n",
+    "\n",
+    "        self.include_agg_to_collab = 100\n",
+    "        self.exclude_agg_to_collab = 78\n",
+    "        self.next(\n",
+    "            self.test_include_exclude_agg_to_collab,\n",
+    "            foreach=\"collaborators\",\n",
+    "            include=[\"include_agg_to_collab\", \"collaborators\"],\n",
+    "        )\n",
+    "\n",
+    "    @collaborator\n",
+    "    def test_include_exclude_agg_to_collab(self):\n",
```

Comment:
By the way, where is the include/exclude logic implemented? Is it the same for `LocalRuntime` and for `FederatedRuntime` ?

Reply:
The main logic for include/exclude is implemented in `openfl/experimental/workflow/utilities/runtime_utils.py`, 
and it is the same for both LocalRuntime and FederatedRuntime.